pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -207,13 +213,15 @@ public BrokerResponse handleRequest(JsonNode request, @Nullable RequesterIdentit
     }
 
     // Set extra settings into broker request
-    if (request.has(TRACE) && request.get(TRACE).asBoolean()) {
+    if (requestParams.isTrace()) {
       LOGGER.debug("Enable trace for request {}: {}", requestId, query);
       brokerRequest.setEnableTrace(true);
     }
-    if (request.has(DEBUG_OPTIONS)) {
-      Map<String, String> debugOptions = Splitter.on(';').omitEmptyStrings().trimResults().withKeyValueSeparator('=')
-          .split(request.get(DEBUG_OPTIONS).asText());
+    Map<String, String> debugOptions = requestParams.getDebugOptions();
+    if (debugOptions != null) {
+//    if (request.has(DEBUG_OPTIONS)) {
```

Comment:
Please remove commented out code if its not needed.

Reply:
Done.